### PR TITLE
un-nerf scanners using a gaussian drop-off and adjusted size dependence

### LINKED ISSUE #7640: Scanners have been nerfed too much
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Due to the changes in https://github.com/endless-sky/endless-sky/pull/6193, successfully scanning another ship with a single human-space scanner is now practically impossible if the other ship is moving. 

### Steps to Reproduce

Manifestations of the issue:
1. The player needs to install 5 or 10 scanners to have any chance of succeeding at missions that require scanning other ships
2. Other ships with scanners that have the "scans" personality and only one scanner like the Republic Gunboat will never successfully scan the player's ship, resulting in those ships simply following the player around forever futilely trying to scan them.

### Expected Behavior

1. Missions that require scanning other ships that move are at least doable with one scanner, and easier with 2 or 3
2. Other ships that try to scan you can succeed at it

### Screenshots

_No response_

### Link to save file

[Morgan Atrasco.txt](https://github.com/endless-sky/endless-sky/files/10037778/Morgan.Atrasco.txt)

In this save I had to equip 10 outfit scanners to succeed at the active mission. With 2, I found it impossible.

### Operating System

Fedora 37

### Game Source

Built from source

### Game Version

commit SHA d925975124dcf3ae67760b161b5e4459056ace5d

### Additional Information

_No response_

### LINKED ISSUE #8272: Crime pays because scanning sucks
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I'm a bit surprised there hasn't been more of a discussion on the changes to scanning, esp. scanning speed. Right now, a simple scan (ie: for the FW storyline, or the Remnant void sprite missions) requires several stacked scanners. 

But the issue goes both ways: in a recent playthrough, the character packed drugs at *every* opportunity. A few missions triggered benign trips to pirate worlds early, and the revenue generated was staggering - because the Navy ships simply can't lock on the way they used to.

The end result is that there are essentially no negative consequences for packing the "good stuff" across the galaxy.

### Expected Behavior

Get busted more often.

### Game Source

Built from source

### Game Version

Continuous build 29 Jan

### PR DESCRIPTION
**Pull request by UnorderedSigh**
**Bugfix:** This un-nerfs scanners so they work again. I expect this fixes #7640 and fixes #8272.

**If parts of this PR are unwanted, I can remove them and still retain the rest. Also, I can switch from gaussian drop-off to linear.**

## Fix Details
Changes the scan formula to un-nerf scanners by using a gaussian drop-off and an adjusted size dependence. This retains the "minigame" from #6193 of ten seconds and cargo/outfit size penalties. It also adds a minimum scan time of 2 seconds. Scanning a Bulk Freighter will take longer than scanning a Sparrow... but you *will* be able to scan it.

1. Minimum scan time of 2 seconds. A Sparrow in Hai space won't be scanned in one frame.
2. More reasonable size dependence: to scan a Bulk Freighter at the same speed as a Sparrow, you only need 6x as many scanners instead of 40x.
3. Scan speed drops with a gaussian ("bell curve") down to 60% of maximum at full range.
4. The numbers are tuned for a Navy surveillance gunship to take about 3 seconds to scan a light to medium warship sized craft in practical conditions.

This is balanced to allow well-designed early-game ships to do illegal missions with proper tactics; while scanning missions don't require dozens of scanners (just 10 seconds of patience).

## Testing Done
Used @Pointedstick's test save with a Splinter and various outfit scanner counts. Spent some time around navy ships to see if they can scan me, and if I could scan them. Everything seems to work about as well as it did before scanners were nerfed into oblivion. However, scan time of bigger ships is still decidedly longer than smaller ones, as intended.

### Free Worlds Scanning Mission

@Pointedstick's test save from #7640 has a scanner-equipped Splinter at the "go scan that Cruiser" mission in the Free Worlds campaign. The mission is doable with one scanner, and easy with three.

[Morgan.Atrasco.txt](https://github.com/endless-sky/endless-sky/files/10549443/Morgan.Atrasco.txt)

### Illegal Missions

Here's another one from @movingpictures: a Heavy Shuttle up to no good, flying through *heavily* patrolled space:

[Lenny.Lightfingers](https://github.com/endless-sky/endless-sky/files/10611805/Lenny.Lightfingers.txt)

Here's four from me: Sparrow, Heavy Shuttle, Clipper, and Blackbird, outfitted for smuggling, all with illegal missions through well-patrolled space, and you don't have enough fuel to reach the destination.

[Aimet Taepip~3029-04-20 Sparrow Test.txt](https://github.com/endless-sky/endless-sky/files/10611903/Aimet.Taepip.3029-04-20.Sparrow.Test.txt)
[Aimet Taepip~3029-04-18 Heavy Shuttle Test.txt](https://github.com/endless-sky/endless-sky/files/10611902/Aimet.Taepip.3029-04-18.Heavy.Shuttle.Test.txt)
[Aimet Taepip~3029-04-18 Clipper Test.txt](https://github.com/endless-sky/endless-sky/files/10611904/Aimet.Taepip.3029-04-18.Clipper.Test.txt)
[Aimet Taepip~3029-04-18 Blackbird Test.txt](https://github.com/endless-sky/endless-sky/files/10611905/Aimet.Taepip.3029-04-18.Blackbird.Test.txt)

I'm able to do all the illegal missions, but I fail if I'm careless. You have to use the old tricks: run away from Navy ships and the Parrot, ramscoop in systems with red stars, refuel by boarding if possible, and never land until your destination. The ships are all quite fast, though the Blackbird relies on afterburners in an emergency.

### Big Ship Illegal Mission and AI Bugs

[Aimet Taepip~3029-04-19 Ranoerek Test.txt](https://github.com/endless-sky/endless-sky/files/10614015/Aimet.Taepip.3029-04-19.Ranoerek.Test.txt)

It generally takes the full ten seconds to scan my Ranoerek.

### CODE REVIEW

**Comment by Zitchas** [comment]
I haven't actually tested this, but I like the sounds of it. Effectivel softening the edge of the scanable range should make it a bit more forgiving.

**Comment by UnorderedSigh** [comment]
> I haven't actually tested this, but I like the sounds of it. Effectivel softening the edge of the scanable range should make it a bit more forgiving.

My testing has been quite limited. People who are in the habit of breaking the law should try this and see how it affects their play style.

File: source/Ship.cpp [hunk 1]
```diff
@@ -2676,32 +2676,20 @@ int Ship::Scan(const PlayerInfo &player)
 	bool activeScanning = false;
 	int result = 0;
 	auto doScan = [&distanceSquared, &startedScanning, &activeScanning, &result]
-			(double &elapsed, const double speed, const double scannerRange,
+			(double &elapsed, const double speed, const double scannerRangeSquared,
 					const double depth, const int event)
 	-> void
 	{
-		if(elapsed < SCAN_TIME && distanceSquared < scannerRange)
+		const int MAX_RANGE_FACTOR = 3 * 3; // three standard deviations AKA triple range
```
Comment (RisingLeaf) [inline source/Ship.cpp:2683]: this is one math operation too much here(micro optimize)

also move a constant like this into the(or an) unnamed namespace.

Comment (UnorderedSigh) [reply source/Ship.cpp:2683]: It is a constant, so it is calculated at runtime. The `3 * 3` is necessary for readability, so the readable sees the important information: 3 times the distance.

Comment (RisingLeaf) [reply source/Ship.cpp:2683]: runtime would be strange XD

still move it to an unnamed namespace for style

Comment (UnorderedSigh) [reply source/Ship.cpp]: It is sitting next to SCAN_TIME now.

File: source/Ship.cpp [hunk 2]
```diff
@@ -65,6 +65,7 @@ namespace {
 	const double MAXIMUM_TEMPERATURE = 100.;
 
 	const double SCAN_TIME = 600.;
+	const int MAX_SCAN_RANGE_FACTOR = 3 * 3; // three standard deviations AKA triple range
```
Comment (RisingLeaf) [inline source/Ship.cpp:68]:
```suggestion
	// Three standard deviations also known as triple range.
	const int MAX_SCAN_RANGE_FACTOR = 3 * 3;
```

**Comment by UnorderedSigh** [comment]
The biggest effect of this on missions is that the maximum scan radius (for that 10 second scan time) is tripled. You only need to get closer than that if you want to scan quickly. Of course, if the target is shooting at you, or trying to jump away, you'll need to be close or have more scanners.

**Review by RisingLeaf — APPROVED** [review]
I consider this good as soon as this last style thing is in.

File: source/Ship.cpp (hunk 2 above)
Comment (UnorderedSigh) [reply source/Ship.cpp]: I added a longer comment explaining the whole thing.

Comment (RisingLeaf) [reply source/Ship.cpp]: even better 👍

**Comment by Amazinite** [comment]
I think the scan range of simply be what it is, not 3x what it is but just mostly effective within the normal range. #7876 already change it so that it's 50% efficiency at max range instead of 0%. That should be sufficient

**Comment by UnorderedSigh** [comment]
> I think the scan range of simply be what it is, not 3x what it is but just mostly effective within the normal range. https://github.com/endless-sky/endless-sky/pull/7876 already change it so that it's 50% efficiency at max range instead of 0%. That should be sufficient

That was already merged. At the top of master, the Navy is still incapable of scanning, and it is painful for the player to do it. Scanning is still quite thoroughly nerfed.

**Comment by UnorderedSigh** [comment]
> but just mostly effective within the normal range

With my changes, scanning is *just mostly effective within the normal range*. The painfully long scan times of modern Endless Sky will only occur beyond that range. This gives you a balance between effective scanning within standard range (as you suggest) and slow but annoying scanning beyond that range (which is enough for mandatory scanning missions).

**Comment by Zitchas** [comment]
To be honest, I see this PR not so much extending the range (as scanning is quite slow in the expanded range), but more in that it prevents the scan from regressing / zeroing out the instant we go beyond the normal range.

**Comment by UnorderedSigh** [comment]
Copying my post from the an issue, people are aware of it.

> I figured out how to solve the other half of the problem and reduce the number of required scanners to a reasonable level so the Navy and players get a reasonable scan time at point-blank range. Amazingly, the solution comes from physical reality!
> 
> There is an error in the calculation of the dependence of scan time on ship size. When you're doing remote sensing, you do not scan the entire three-dimensional structure of the target to determine its contents. You scan the cross-section, and the quantity in each pixel is calculated from the magnitude of the return or emission (ie. pixel value). The dependence should be `1/pow(depth,0.6666)` instead of `1/depth`. Also, the `max(10., cargo)` should be more like `max(40., cargo)`. (Credit goes to @NomadicVolcano for explaining the science to me.)
> 
> That'll leave you with about 6x as many scanners to scan a Bulk Freighter than a Sparrow, rather than 40x. I'll do a PR later to adjust that formula.

**Comment by movingpictures** [comment]
Tested this branch this morning. Took a helluva time to trigger drug missions, damnit, but the two times poor Lenny Lightfingers tried to do something illegal, he got by once, and then.... Reps/navy jumped on his innocuously named "Cheech and Chong II" twice in two jumps.

pirate

[Lenny Lightfingers.txt](https://github.com/endless-sky/endless-sky/files/10611636/Lenny.Lightfingers.txt)

My immediate gut reaction: this PR doesn't find the Goldilocks zone.

**Comment by UnorderedSigh** [comment]
> "Warning: color "ship target pointer unfriendly" is referred to, but not fully defined."

This is unrelated to my changes.

**Comment by UnorderedSigh** [comment]
> My immediate gut reaction: this PR doesn't find the Goldilocks zone.

My change to the `depth` variable (ship size adjustment) will counteract this. It is impossible to tune the two independently.

Should I combine the ship size adjustment into this PR?  I don't see any other way...

**Comment by UnorderedSigh** [comment]
>  Navy nabbed poor Lenny Lightfingers and his perfectly not suspicious heavy shuttle "Cheech and Chong II" twice in successive jumps

If you're using a small slow ship (heavy shuttle), you *should* get caught. That's the system working as designed. However, I do think it'll be too easy to get caught with the numbers as they currently are. I think we need something like this:

- 10 seconds to scan the cargo of a Bulk Freighter at point blank range with one Cargo Scanner (efficiency 5)
- 5 seconds to scan the outfits of a Bactrian at point blank range with one Outfit Scanner (efficiency 15)

**Comment by movingpictures** [comment]
> If you're using a small slow ship (heavy shuttle), you _should_ get caught. 

Lenny, in preparation for a life of depravity, tossed Deep Sky atomics into his shuttle, giving a speed of 486.

>     * 10 seconds to scan the cargo of a Bulk Freighter at point blank range with one Cargo Scanner (efficiency 5)
> 
>     * 5 seconds to scan the outfits of a Bactrian at point blank range with one Outfit Scanner (efficiency 15) 

I'd say that one scan came maybe a second after flashing into the system, the other was almost instantaneous.

**Comment by UnorderedSigh** [comment]
> I'd say that one scan came maybe a second after flashing into the system, the other was almost instantaneous.

Can you give me a save file for testing this?

**Comment by movingpictures** [comment]
> > I'd say that one scan came maybe a second after flashing into the system, the other was almost instantaneous.
> 
> Can you give me a save file for testing this?

'Twas in the original reply, but:

[Lenny Lightfingers.txt](https://github.com/endless-sky/endless-sky/files/10611805/Lenny.Lightfingers.txt)

**Comment by UnorderedSigh** [comment]
Sorry everyone, but I had to include my changes to the ship size dependence ("depth" formula) or the illegal missions were impossible. The two are well-balanced now. Illegal missions are difficult, but doable with proper tactics. The free worlds scanning mission test from Pointedstick is easy with three Outfit Scanners, and doable with one.

**Comment by UnorderedSigh** [comment]
Could someone please re-request a review from @Pointedstick? When I asked RisingLeaf to re-review, it un-requested Pointedstick.

**Comment by movingpictures** [comment]
So I ran the altered/tweaked version, and now I think it's moved too far to the other side of the spectrum.

Sitting parked, it took a navy gunboat just shy of seven seconds to catch the contraband.

My thought: three seconds seems about right. FYI,

**Comment by UnorderedSigh** [comment]
Okay, I have tweaked it again:

- 5 seconds to cargo scan a Bulk Freighter at point-blank range
- 3 seconds to outfit scan a Bactrian at point-blank range

The time it'll actually take to scan will be longer because the ship won't be at point-blank range.

**Comment by UnorderedSigh** [comment]
It takes quite a while to scan my Rano'erek, which has about the same outfit space as a Bactrian and cargo space as a Bulk Freighter. The gunboats are unable to stay within scanner range, so they have much longer than the 3-5 second estimates. Also, they'll do an outfit scan first, and then wander off, skipping the cargo scan.

**Comment by UnorderedSigh** [comment]
It would be great if @Ferociousfeind could look at this PR, since they wrote the formulation I am presently reformulating.

**Comment by Zitchas** [comment]
Right now, I think we have some problems regarding what exactly we're targeting for scan times. For someone with a single human scanner, I don't think it's unreasonable for even a perfect parked-on-top scan of a Carrier or Bulk Freighter or other similarly huge spaceship to take a full ten seconds. A single stand alone scanner is the minimum viable setup to get a scan done, and comes with the assumption that if someone wants to scan something big they'll probably need more than one. A single outfit is the "standard" for scanning *average* sized human ships, not the biggest human ships.

**Comment by UnorderedSigh** [comment]
> For someone with a single human scanner, I don't think it's unreasonable for even a perfect parked-on-top scan of a Carrier or Bulk Freighter or other similarly huge spaceship to take a full ten seconds.

It takes a full ten seconds for a gunship to do that scan.

**Comment by UnorderedSigh** [comment]
> A single outfit is the "standard" for scanning average sized human ships, not the biggest human ships.

That is how @movingpictures and I have been tuning it. I'm just using the biggest human ships in my equations, since big numbers are easier to work with.

The test ships are:

- Blackbird
- Heavy Shuttle
- Clipper
- Sparrow

**Comment by UnorderedSigh** [comment]
> It takes a full ten seconds for a gunship to do that scan.

Clarification. The gunship can't sit on top of it. That's why it takes ten seconds.

**Comment by Ferociousfeind** [comment]
I'm wary of scan times coming so low (the "finger hovering over the afterburner button" sentiment is a good argument to me) and risking making smuggling very difficult again... I'm also having trouble replicating the formula in desmos to play with, I'm not sure the operations I have are right, but I assume the numbers you espouse are correct. If they're correct, 5 seconds to scan a really big ship is a somewhat concerning amount in my opinion. But whatever, I haven't tested running away from gunboats recently, what do I know?

I'll give it my official "wait why is this my responsibility now" seal of approval

**Comment by UnorderedSigh** [comment]
> If they're correct, 5 seconds to scan a really big ship is a somewhat concerning amount in my opinion

It's not that short, actually. If it weren't for the 10-second limit, the AI ships would take 20–30 seconds to scan a big ship's cargo or outfits, and another 20–30 seconds to scan the other category.

> making smuggling very difficult again
> finger hovering over the afterburner button

Smuggling should be difficult in a ship not designed for it. With proper tactics, and a well-designed ship, you should be able to smuggle reliably. I'm able to do all four of my saves with near 100% reliability, unless I get careless. Right now, any ship can smuggle without worrying about getting caught, and that is a completely broken situation.

There's no point in having a mission be illegal if it is impossible to get caught.

**Comment by Quantumshark** [comment]
I think the best way to maintain a balance between making scanning not overly painful for the player while still allowing the player to evade scans would be to increase the strength and variety of anti-scanning outfits.
Right now, the only anti-scanning outfits we have provide a modest amount of `scan interference`, which gives a small random chance for a scan to fail.
I would propose that we add a second anti-scanning attribute (`scan opacity`)? that makes scanning a ship slower, and also add a `scan penetration` attribute (which could be separate for cargo vs outfit scans, too) that scanners can have that reduces the effect of `scan interference` but not `scan opacity`.
This would create an environment in which a player could combine `scan opacity` outfits with their own piloting skills in order to evade scans, or, depending on what scanners they were expecting to be trying to evade, could instead stack lots of `scan interference` and hope they get lucky.
This would also create additional scope for variety in scanners, since they'd have one more attribute to play with.

**Comment by UnorderedSigh** [comment]
While @Quantumshark's ideas are excellent, and would bring wonderful variety to the game, they would only do that if it were actually possible to be scanned. Presently, for a medium warship to get scanned, you really have to *try* to get scanned, and having a large ship is a "get out of jail free" card due to AI bugs. The scanning formulae must be corrected before new scan-blocking stats and outfits could be useful.

**Comment by Quantumshark** [comment]
> While @Quantumshark's ideas are excellent, and would bring wonderful variety to the game, they would only do that if it were actually possible to be scanned. Presently, for a medium warship to get scanned, you really have to _try_ to get scanned, and having a large ship is a "get out of jail free card" due to AI bugs. The scanning formulae must be corrected before new scan-blocking stats and outfits could be useful.

You are absolutely right, and I am 100% in favour of these changes - my point was that with changes like the ones I suggested, we could make scanning much faster and make the AI much better at it _without_ making anti-scan gameplay impossible or excessively difficult.

**Comment by UnorderedSigh** [comment]
A fix for one of the AI bugs I ran into with big ships is here: #8323. With that bug fix, scanning ships will stay on top of their target instead of moving erratically around it. I expect, with the two fixes combined, scanners will be too strong, so we'll have to adjust `SCAN_CARGO_FACTOR` and `SCAN_OUTFIT_FACTOR` to compensate.

**Comment by UnorderedSigh** [comment]
I've updated the formulae in this PR to work well with the improved movement pattern of scanning ships in #8323. As expected, I had to weaken the scanners slightly to compensate for the ships moving competently.

**Comment by UnorderedSigh** [comment]
Last tested and known to work at Sun, 19 Feb 2023 04:26:59 UTC

**Comment by UnorderedSigh** [comment]
Last tested and confirmed to work at Sat, 04 Mar 2023 17:53:13 UTC.

However, as other scanner changes are merged, the parameters in this branch may need to be re-tuned to get whatever target scan time people want.

**Comment by UnorderedSigh** [comment]
Last tested and known to work Sat, 25 Mar 2023 22:35:22.

**Comment by mOctave** [comment]
This is something that seems to be pretty important, with the complaints people are making regarding scanning. It also seems to have fallen in the hole of PRs that will never get merged because they got too old.

I feel like this needs some extra attention, so it will be ready for whenever the next release is. Otherwise, it will just become another great change that never got implemented and eventually got closed due to mounting piles of merge conflicts and other PRs.

**Comment by UnorderedSigh** [comment]
> mounting piles of merge conflicts

I update my PRs every few weeks to make sure there are no major conflicts, and to make sure they still work. You will see "last tested and known to work" every few weeks in the PRs I actively maintain.

**Comment by UnorderedSigh** [comment]
Last tested and known to work Sun, 16 Apr 2023 23:49:57 UTC

**Comment by UnorderedSigh** [comment]
I had to tweak things a bit for them to work well with the latest scanning logic changes.

Last tested and known to work Fri, 12 May 2023 04:38:47 UTC.

**Comment by UnorderedSigh** [comment]
I have added a *minimum scan time* to the formula. No matter how small your ship is, nor how advanced your opponent is, you will take a minimum of two seconds to scan. Maximum is still 10.

**Comment by UnorderedSigh** [comment]
The `CI / Ubuntu (20.04, GL) (pull_request)` test keeps failing due to sound errors. This has nothing to do with my PR. I'll try re-running the tests later.

```
  ALSA lib .././../src/v1.2.6.1-b673b0105e.clean/src/confmisc.c:855:(parse_card) cannot find card '0'
  ALSA lib .././../src/v1.2.6.1-b673b0105e.clean/src/conf.c:5178:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
  ALSA lib .././../src/v1.2.6.1-b673b0105e.clean/src/confmisc.c:422:(snd_func_concat) error evaluating strings
  ALSA lib .././../src/v1.2.6.1-b673b0105e.clean/src/conf.c:5178:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
  ALSA lib .././../src/v1.2.6.1-b673b0105e.clean/src/confmisc.c:1334:(snd_func_refer) error evaluating name
  ALSA lib .././../src/v1.2.6.1-b673b0105e.clean/src/conf.c:5178:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
  ALSA lib .././../src/v1.2.6.1-b673b0105e.clean/src/conf.c:5701:(snd_config_expand) Evaluate error: No such file or directory
  ALSA lib ../.././../src/v1.2.6.1-b673b0105e.clean/src/pcm/pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM default
  Warning: audio could not be initialized. No audio will play.
  Test failed: watchdog timeout
```

**Comment by UnorderedSigh** [comment]
The integration tests were assuming a ship could be scanned instantaneously. I increased the scanning test watchdog limit to account for the 2 second minimum scan time. It took some debugging and a few red herrings to figure this out, hence the odd sequence of commits.

Last tested and known to work Mon, 29 May 2023 21:39:46 UTC

**Comment by UnorderedSigh** [comment]
Last tested and known to work Fri, 02 Jun 2023 04:57:32 UTC.

**Comment by warp-core** [comment]
@UnorderedSigh I've opened a PR (https://github.com/UnorderedSigh/endless-sky/pull/35) to this branch to bring it up to date and:
* Resolve the conflict which was with the addition of the cargo and outfit scan opacity attributes.
* Remove the increased scan range. Based on the discussions I've seen, it seems like not losing scan progress while out of range (as long as you don't switch target) is preferred to accruing scan progress well outside the normal range, and we have that now.
* Add some comments to describe the maths.
* Fix style in one or two places.

**Comment by UnorderedSigh** [comment]
An integration test is failing after @warp-core's changes, and I'm able to reproduce this on my test machine.

Also, I don't know if this PR fixes or breaks scanning, after the extended range was removed. Extending the range was *the reason* scanning missions were doable again. I'll test all of the cases in the PR description.

**Comment by UnorderedSigh** [comment]
Something is wrong with the scan range. Ships are able to scan me from *way* past their scan range now.

**Comment by UnorderedSigh** [comment]
Found the cause of both problems. After I merged warp_core's changes, ships were only able to scan when they were *beyond* scan range. This was due to a `<` that should have been a `>`. Let's see if the tests go through on all platforms now...

**Comment by UnorderedSigh** [comment]
I've tried the test cases in the PR description, and they all seem reasonable. I was able to do the Morgan.Atrasco mission with one scanner.

**Comment by UnorderedSigh** [comment]
Last tested and known to work Fri, 18 Aug 2023 02:36:14 UTC.

At least, it works to my satisfaction. Whether it is "good enough" is entirely subjective. The smuggling ships I tested were ideally suited for smuggling. Try smuggling in a slow warship, and you will get caught. In my mind, that's how it should be.

The "great equalizer" is the two second minimum scan time. As long as your ship can get away in two seconds, you can smuggle anything. There are many ways to design a ship around that: afterburners, fast engines, scram drive, cloak, etc. They all have tradeoffs. If you don't design around this, you will get caught.

**Pull request merged**